<a href="https://colab.research.google.com/github/aneuni/ITB-2022/blob/main/2020%EB%85%84%EB%8F%84_%EA%B1%B4%EA%B0%95%EA%B2%80%EC%A7%84%EC%A0%95%EB%B3%B4_%EC%83%98%ED%94%8C_%EC%8B%9C%EA%B0%81%ED%99%94_1914838_%EC%95%88%EC%9D%80%EC%9D%B4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#기본 과정

In [1]:
from matplotlib import pyplot as plt
import pandas as pd
from pandas import Series, DataFrame
import numpy as np
from matplotlib import pyplot as plt
%matplotlib inline

# 한글 폰트 설정 (윈도우 Malgun Gothic,  리눅스&Mac AppltGothic)
import matplotlib.font_manager as fm
!sudo apt-get install -y fonts-nanum #Isudo = super user do
!sudo fc-cache -fv
!rm ~/.cache/matplotlib -rf
plt.rc('font', family='NanumBarunGothic') 

Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following package was automatically installed and is no longer required:
  libnvidia-common-460
Use 'sudo apt autoremove' to remove it.
The following NEW packages will be installed:
  fonts-nanum
0 upgraded, 1 newly installed, 0 to remove and 29 not upgraded.
Need to get 9,604 kB of archives.
After this operation, 29.5 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 fonts-nanum all 20170925-1 [9,604 kB]
Fetched 9,604 kB in 1s (10.9 MB/s)
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 1.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (This frontend requires a controlling tty.)
debconf: falling back to frontend: Teletype

In [2]:
#구글드라이브 연결
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
#파일불러오기
data = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/IT와비즈니스혁신/data/health_data.csv')

In [5]:
data

기준년도 가입자일련번호 시도코드 성별코드 연령대코드(5세단위) 신장(5Cm단위) 체중(5Kg 단위)  허리둘레 시력(좌)  \
0      2020       1   36    M           9       165         60  72.1   1.2   
1      2020       2   27    F          13       150         65    81   0.8   
2      2020       3   11    F          12       155         55    70   0.6   
3      2020       4   31    M          13       160         70  90.8     1   
4      2020       5   41    F          12       155         50  75.2   1.5   
...     ...     ...  ...  ...         ...       ...        ...   ...   ...   
9997   2020    9998   45    M          12       160         80    93   0.6   
9998   2020    9999   27    F          15       155         55    78   0.6   
9999   2020   10000   45    F          13       160         75   100   0.5   
10000     0       0    0    0           0         0          0     0     0   
10001    0%      0%   0%   0%          0%        0%         0%    0%    0%   

      시력(우)  ... 혈청크레아티닌 (혈청지오티)AST (혈청지오티)ALT 감마지티피    흡연상태 음주여부 구강검진수검여부  \
0       1.5  ...     1.1         21         27    21      No   No        0   
1       0.8  ...     0.5         18         15    15      No   No        0   
2       0.7  ...     0.7         27         25     7      No   No        0   
3         1  ...     1.2         65         97    72      No   No        1   
4       1.2  ...     0.7         18         17    14      No   No        0   
...     ...  ...     ...        ...        ...   ...     ...  ...      ...   
9997    0.6  ...       1         22         27    35      No   No        0   
9998      1  ...       1         26         24    21      No   No        0   
9999    0.5  ...     0.6         32         24    30  NotNow  Yes        0   
10000     0  ...      88         88         88    88       2    1        0   
10001    0%  ...      1%         1%         1%    1%      0%   0%       0%   

      치아우식증유무    치석     데이터공개일자  
0         NaN   NaN  12/29/2021  
1         NaN   NaN  12/29/2021  
2         NaN   NaN  12/29/2021  
3           0     0  12/29/2021  
4         NaN   NaN  12/29/2021  
...       ...   ...         ...  
9997      NaN   NaN  12/29/2021  
9998      NaN   NaN  12/29/2021  
9999      NaN   NaN  12/29/2021  
10000    6749  6749           0  
10001     67%   67%          0%  

[10002 rows x 31 columns]

엑셀로 결측치 개수 확인 결과(10000행: 결측치 개수 / 10001행: 결측치 비율)
총콜레스테롤, 트리글리세라이드, HDL콜레스테롤, LDL콜레스테롤, 치아유식증유무, 치석은 결측치가 60% 이상임 (시각화 자료로 사용하지 않을 예정)


In [7]:
# 데이터 컬럼 정보
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10002 entries, 0 to 10001
Data columns (total 31 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   기준년도         10002 non-null  object
 1   가입자일련번호      10002 non-null  object
 2   시도코드         10002 non-null  object
 3   성별코드         10002 non-null  object
 4   연령대코드(5세단위)  10002 non-null  object
 5   신장(5Cm단위)    10002 non-null  object
 6   체중(5Kg 단위)   10002 non-null  object
 7   허리둘레         10002 non-null  object
 8   시력(좌)        10002 non-null  object
 9   시력(우)        10002 non-null  object
 10  청력(좌)        10002 non-null  object
 11  청력(우)        10002 non-null  object
 12  수축기혈압        9914 non-null   object
 13  이완기혈압        9914 non-null   object
 14  식전혈당(공복혈당)   9914 non-null   object
 15  총콜레스테롤       4046 non-null   object
 16  트리글리세라이드     4046 non-null   object
 17  HDL콜레스테롤     4046 non-null   object
 18  LDL콜레스테롤     3979 non-null   object
 19  혈색소          9914 non-nul

In [6]:
# 데이터 통계 정보
data.describe()

기준년도 가입자일련번호   시도코드   성별코드 연령대코드(5세단위) 신장(5Cm단위) 체중(5Kg 단위)   허리둘레  \
count   10002   10002  10002  10002       10002     10002      10002  10002   
unique      3   10002     19      4          12        16         22    425   
top      2020       1     41      M          11       165         60     80   
freq    10000       1   2495   5138        1798      1837       1618    411   

        시력(좌)  시력(우)  ... 혈청크레아티닌 (혈청지오티)AST (혈청지오티)ALT 감마지티피   흡연상태   음주여부  \
count   10002  10002  ...    9914       9914       9914  9914  10000  10001   
unique     19     19  ...      43        148        171   316      5      4   
top         1      1  ...     0.8         21         17    17     No    Yes   
freq     2049   2082  ...    1898        599        497   394   6231   5819   

       구강검진수검여부 치아우식증유무    치석     데이터공개일자  
count     10002    3253  3253       10002  
unique        3       4     5           3  
top           0       0     1  12/29/2021  
freq       6750    2681  1540       10000  

[4 rows x 31 columns]